eng/common/TestResources/New-TestResources.ps1

```diff
@@ -125,366 +125,374 @@ $exitActions = @({
     }
 })
 
-trap {
-    # Like using try..finally in PowerShell, but without keeping track of more braces or tabbing content.
-    $exitActions.Invoke()
-}
-
-# Enumerate test resources to deploy. Fail if none found.
-$repositoryRoot = "$PSScriptRoot/../../.." | Resolve-Path
-$root = [System.IO.Path]::Combine($repositoryRoot, "sdk", $ServiceDirectory) | Resolve-Path
-$templateFileName = 'test-resources.json'
-$templateFiles = @()
-# Azure SDK Developer Playground
-$defaultSubscription = "faa080af-c1d8-40ad-9cce-e1a450ca5b57"
-
-Write-Verbose "Checking for '$templateFileName' files under '$root'"
-Get-ChildItem -Path $root -Filter $templateFileName -Recurse | ForEach-Object {
-    $templateFile = $_.FullName
-
-    Write-Verbose "Found template '$templateFile'"
-    $templateFiles += $templateFile
-}
-
-if (!$templateFiles) {
-    Write-Warning -Message "No template files found under '$root'"
-    exit
+New-Variable -Name 'initialContext' -Value (Get-AzContext) -Option Constant
```

Comment:
Why not $initialContext = Get-AzContext?

Reply:
Extra protection to make sure it's not overwritten accidentally (see `-Option Constant`).

Reply:
Why is this particular variable warrant the extra protection and not others?